03-transactions.md

```diff
@@ -710,6 +719,102 @@ at each bucket is a prefix of the desired index.
 
 # Appendix A: Expected Weights
 
+## Expected Weight of the Funding Transaction (v2 Channel Establishment)
+
+The *expected weight* of a funding transaction is calculated as follows:
+
+      inputs: 40 bytes + var_int + `scriptlen`
+        - previous_out_point: 36 bytes
+          - hash: 32 bytes
+          - index: 4 bytes
+        - var_int: ? bytes (dependent on `scriptlen`)
+        - script_sig: `scriptlen`
+        - witness <---- Cost for "witness" data calculated separately.
+        - sequence: 4 bytes
+
+      non_funding_outputs: 8 bytes + var_int + `scriptlen`
+        - value: 8 bytes
+        - var_int: ? bytes (dependent on `scriptlen`)
+        - script: `scriptlen`
+
+      funding_output: 43 bytes
+        - value: 8 bytes
+        - var_int: 1 byte
+        - script: 34 bytes
+          - OP_0: 1 byte
+          - PUSHDATA(32-byte-hash): 33 bytes
+
+Multiplying non-witness data by 4 results in a weight of:
+
+      // transaction_fields = 10 (version, input count, output count, locktime)
+      // segwit_fields = 2 (marker + flag)
+      // funding_transaction = 43 + num_inputs * 40 + num_outputs * 8
+      //                       + sum(scriptlen) + sum(var_ints) +
+      funding_transaction_weight = 4 * (funding_transaction + transaction_fields) + segwit_fields
+
+      witness_weight = sum(witness_len)
+
+      overall_weight = funding_transaction_weight + witness_weight
+
+### Calculating Fees for Collaborative Transaction Construction
+
+Every participant in a collaborative transaction covers the fees for
+their own inputs and outputs. The initiator also provides funds to
+cover for the common transaction fields.
+
+Upon successful exchange of `tx_complete` messages, each peer must at
+least cover a minimum estimated fee. Here is how to calculate the minimum
+fee for each participant.
+
+The minimum witness weight for an input is 110.
```

Comment:
It's actually 107, not 110: https://bitcoin.stackexchange.com/questions/100159/what-is-the-size-and-weight-of-a-p2wpkh-input/100160

I've hit that in one of my tests, where only the funder contributes with a single p2wpkh input.

Reply:
Right, the minimum weight for a p2wpkh is 107; I picked 110 for reasons I don't remember now, something arbitrary about how sometimes scripts are multisigs. Moving it down to 107 is probably the right move here.

Reply:
updated!